--- OpenCVFindMKL.cmake	2018-10-07 22:42:46.618955024 +0000
+++ new_OpenCVFindMKL.cmake	2018-10-07 23:12:36.267540713 +0000
@@ -97,22 +97,11 @@
 if(${MKL_VERSION_STR} VERSION_GREATER "11.3.0" OR ${MKL_VERSION_STR} VERSION_EQUAL "11.3.0")
     set(mkl_lib_find_paths
         ${MKL_ROOT_DIR}/lib
-        ${MKL_ROOT_DIR}/lib/${MKL_ARCH} ${MKL_ROOT_DIR}/../tbb/lib/${MKL_ARCH})
+        ${MKL_ROOT_DIR}/lib/${MKL_ARCH} /opt/intel/composerxe/linux/tbb/lib/intel64/gcc4.7/) # We should use ${TBB_ENV_LIB} instead
 
     set(mkl_lib_list "mkl_intel_${MKL_ARCH_SUFFIX}")
 
-    if(MKL_WITH_TBB)
-        list(APPEND mkl_lib_list mkl_tbb_thread tbb)
-    elseif(MKL_WITH_OPENMP)
-        if(MSVC)
-            list(APPEND mkl_lib_list mkl_intel_thread libiomp5md)
-        else()
-            list(APPEND mkl_lib_list mkl_gnu_thread)
-        endif()
-    else()
-        list(APPEND mkl_lib_list mkl_sequential)
-    endif()
-
+    list(APPEND mkl_lib_list mkl_tbb_thread tbb)
     list(APPEND mkl_lib_list mkl_core)
 else()
     message(STATUS "MKL version ${MKL_VERSION_STR} is not supported")
